lpshipit.py

```diff
@@ -143,153 +157,149 @@ def lpshipit(directory, source_branch, target_branch, mp_owner, debug):
         print('Debug: Launchad returned {} merge proposals'.format(len(mps)))
     mp_summaries = summarize_git_mps(mps)
 
-    if mp_summaries:
+    if not mp_summaries:
+        print("You have no Merge Proposals in either "
+              "'Needs review' or 'Approved' state")
+        sys.exit(1)
+
+    if not directory:
+        directory = os.getcwd()
 
-        def urwid_exit_on_q(key):
-            if key in ('q', 'Q'):
-                raise urwid.ExitMainLoop()
+    if not os.path.isdir(directory):
+        print("'%s' is not a directory" % directory)
+        sys.exit(1)
 
-        def urwid_exit_program(button):
+    if not mp_summaries:
+        print("You have no Merge Proposals in either "
+              "'Needs review' or 'Approved' state")
+        sys.exit(1)
+
+    # filter MPs that aren't related to the chosen directory based on the
+    # URLs of git repo remotes
+    repo = git.Repo(directory)
+    remotes = convert_remotes_to_lp_urls(repo)
+    mp_summaries = filter(
```

Comment:
ordering feels weird:

we get MP summaries, check things, filter, do the same check...

I think this should be re-ordered 

1. check directory
2. get proposals
3. filter proposals
4. check for summaries

Reply:
I've moved directory-related stuff to the top.